### PR TITLE
https://github.com/phpseclib/phpseclib/issues/1113

### PR DESCRIPTION
**Pull request by jasekiw**
Add compatibility with phpseclib 2.0.3 - 2.0.4.

### CODE REVIEW

File: lib/openpgp_crypt_rsa.php [hunk 1]
```diff
@@ -9,6 +9,7 @@
 // From http://phpseclib.sourceforge.net/
 use phpseclib\Crypt\RSA as Crypt_RSA;
 use phpseclib\Math\BigInteger as Math_BigInteger;
+use phpseclib\Math\BigInteger;
```
Comment (singpolyma) [inline lib/openpgp_crypt_rsa.php]: Do we need both of these use lines for the same module?

File: lib/openpgp_crypt_rsa.php [hunk 2]
```diff
@@ -245,8 +246,18 @@ static function convert_key($packet, $private=false) {
...
-      if($packet->key['p'] && $packet->key['q']) $rsa->primes = array($packet->key['p'], $packet->key['q']);
-      if($packet->key['u']) $rsa->coefficients = array($packet->key['u']);
+        /**
+         * @see https://github.com/phpseclib/phpseclib/issues/1113
+         * Primes and coefficients now use BigIntegers.
+         **/
+        //set the primes
+        if($packet->key['p'] && $packet->key['q'])
+            $rsa->primes = array(
+                1=> new BigInteger($packet->key['p'], 256),
```
Comment (singpolyma) [inline lib/openpgp_crypt_rsa.php]: Space before `=>` and specify key for both array items?

File: lib/openpgp_crypt_rsa.php [hunk 3]
```diff
@@ -245,8 +246,18 @@ static function convert_key($packet, $private=false) {
...
+                new BigInteger($packet->key['q'], 256)
+            );
+        // set the coefficients
+        if($packet->key['u']) $rsa->coefficients = array(2 => new BigInteger($packet->key['u']));
```
Comment (singpolyma) [inline lib/openpgp_crypt_rsa.php]: Does this constructor need to be passed the 256 like the others above?

**Comment by singpolyma** [comment]
Thanks so much for working on this!

File: lib/openpgp_crypt_rsa.php (hunk 1 above)
Comment (jasekiw) [inline lib/openpgp_crypt_rsa.php]: No we do not. Removing that now.

File: lib/openpgp_crypt_rsa.php (hunk 3 above)
Comment (jasekiw) [inline lib/openpgp_crypt_rsa.php]: I'm not too sure what the bases should be. That is the fix that was in the issue https://github.com/phpseclib/phpseclib/issues/1113. Should we take out the base out of both or add 256 to both? How will that affect the encryption?

File: lib/openpgp_crypt_rsa.php (hunk 2 above)
Comment (jasekiw) [inline lib/openpgp_crypt_rsa.php]: adding that now.

**Comment by jasekiw** [comment]
That's strange. I did not get those errors on php 7.1 and 7.0. Could you test the commit on your side?

File: lib/openpgp_crypt_rsa.php (hunk 3 above)
Comment (jasekiw) [inline lib/openpgp_crypt_rsa.php]: looks like the tests point out that it does. Not sure why the tests didn't catch that on my machine.

**Comment by singpolyma** [comment]
Weird that it's only failing on PHP 7+

**Comment by jasekiw** [comment]
apparently mcrypt_module_get_algo_key_size(), mcrypt_module_get_algo_block_size, and  mcrypt_get_iv_size is deprecated. I just configured my php to get that error.  I can look into alternatives for those methods. For some reason the 7.0 stalled. I think that may have been travis. Is there a way restart the tests?

**Comment by singpolyma** [comment]
I've restarted the stuck 7.0 -- if it passes we can nix the adding of 7.1 to the tests for the moment and merge this (maybe squash the commits?) and re-add 7.1 after the deprecations are figured out

**Comment by jasekiw** [comment]
sounds good. I started looking into the issue and it seems pretty rooted in. The whole MCryptWrapper would have to be taken out. How can I squash the commits? Would I force push over my current and combine it into one commit and apply it to the pull request?

**Comment by singpolyma** [comment]
Yeah. `git rebase -i HEAD~3` and use the fixup option to squash. Then force push :)

**Comment by jasekiw** [comment]
I squashed all my commits and removed 7.1 testing

**Comment by jasekiw** [comment]
travis seemed to have stalled twice.

**Pull request merged**